### ISSUE TITLE
TcBlackCore can not be build

### ISSUE THREAD

**Issue by SankariNL** [Roald87/TcBlack#57]
When trying to build the TcBlackCore project I get the following error: 

Unable to get MD5 checksum for the key file "TcBlackCoreSign.pfx". Could not find file 'C:\Users\_username_\source\repos\_username_\TcBlack\src\TcBlackCore\TcBlackCoreSign.pfx'.

This file is mentioned in the solution explorer, but is not included in the repo.

After removing the file in the solution explorer the solution could be build.

**Comment by PTKu** [comment]
@SankariNL you will need to create your own ```pfx``` file. Open ```TcBlackCore``` project properties navigate to signing. Roll out the combo and select new...
![image](https://user-images.githubusercontent.com/61538034/113766627-ecf11380-971d-11eb-8063-493535c94604.png)

## NOTE **You will need to set a password otherwise you will get ```snk``` and not ```pfx``` file**


![image](https://user-images.githubusercontent.com/61538034/113767229-b9fb4f80-971e-11eb-86a8-cc1c29d9d4aa.png)

**Comment by Roald87** [comment]
Thanks for answering @PTKu! I didn't even realise the signature was added and didn't know if it was really necessary. After some Googling I found that for some applications it makes sense for others not. But I'm still unsure whether it makes sense for _TcBlack_. What's your take @PTKu ?

**Comment by PTKu** [comment]
That's a good question. You're right there are many opinions out there whether sign or not to sign :). I do not have definite answer for that. We do sign all assemblies that are going public. 
I am not sure about this, but you may need to sign your assemblies when you use them in VS extension, I vaguely recall that was the requirement. 
Rule of thumb: if you are not sure just sign. From time to time you will get intro troubles with sign assemblies (testing internal members require some additional work beside ```InternalVisibleTo(assemblyName)``` attribute, in some situation you may have a third party assembly that has not signed version available and you need to workaround that...). 
Anyways it adds some layer of security to your application, I guess it's worth the pain:)

**Comment by Roald87** [comment]
@PTKu Thanks for the clarification! Then I'll keep it.

**Comment by SankariNL** [comment]
Hello guys, thanks for the feedback!

I've added some extra documentation about this, see: https://github.com/Roald87/TcBlack/pull/65#issue-623130396

Also for some reason I trying to contribute using Visual Studio 2019. This works fine for the most part, there is just 1 reference that is not available in Visual Studio 2019, and also not on nuget.org. I can contribute by manually changing the reference, but not commiting it. I've also included this in the documentation.
